firmware/common/firmware_info.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 Great Scott Gadgets <info@greatscottgadgets.com>
+ *
+ * This file is part of HackRF.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#ifndef FIRMWARE_INFO_H
+#define FIRMWARE_INFO_H
+
+#include <stdint.h>
+
+struct firmware_info_t {
+	char magic[8];
+	uint32_t struct_version;
+	uint32_t supported_platform;
+	char version_string[32];
```

Comment:
Could you please add a field that indicates whether or not the firmware was compiled with DFU_MODE?

Reply:
Done! I've reduced the size of the `struct_version` field and added `dfu_mode` next to it, to keep everything on nice boundaries.